Tests/SwiftParserTest/MemberRefTests.swift

```diff
@@ -0,0 +1,226 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import SwiftParser
+import XCTest
+
+final class MemberTypeTests: XCTestCase {
+  func testKeyword() {
+    AssertParse(
+      "MyType.class",
+      { TypeSyntax.parse(from: &$0) },
+      substructure: Syntax(MemberTypeIdentifierSyntax(
+        baseType: SimpleTypeIdentifierSyntax(
+          name: .identifier("MyType")
+        ),
+        period: .period,
+        name: .identifier("class")
+      )))
+  }
+
+  func testMissing() {
+    AssertParse(
+      "MyType.1️⃣",
+      { TypeSyntax.parse(from: &$0) },
+      substructure: Syntax(MemberTypeIdentifierSyntax(
+        baseType: SimpleTypeIdentifierSyntax(
+          name: .identifier("MyType")
+        ),
+        period: .periodToken(),
+        name: .identifier("", presence: .missing)
+      )),
+      diagnostics: [DiagnosticSpec(message: "expected member type in member type")])
```

Comment:
Uh, that diagnostic message doesn't seem ideal. Do you know if we can improve it by updating some `name_for_diagnostic`s? Doesn't have to be this PR.